### PR TITLE
Veliseev/expo router

### PR DESCRIPTION
**Pull request by veliseev93**

### CODE REVIEW

**Comment by ipakhomov** [comment]
@veliseev93 thanks for the update!

Could you please apply some extra enhancements?
1. Remove the `appNavigationService` and `appNavigationEpics`
2. Turn on `"strictNullChecks": true` in `tsconfig.json` and fix typescript issues
3. Update `devDependencies` related to linting and formatting

**Review by ipakhomov — CHANGES_REQUESTED** [review]
See required changes above

**Comment by ipakhomov** [comment]
@veliseev93 let's also migrate the project to use `expo-image` instead of [current image implementation](https://github.com/RonasIT/react-native-starter/blob/main/libs/shared/ui/ui-kit/external-image/component.tsx#L4)

**Comment by veliseev93** [comment]
@veliseev93 Done, please check

File: libs/shared/ui/ui-kit/external-image/component.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ interface Props {
 
 export function ExternalImage({ uri, style = {}, onPress }: Props): ReactElement {
   const renderedImage =
-    Platform.OS === 'web' ? <Image source={{ uri }} style={style} /> : <CachedImage uri={uri} style={style} />;
+    Platform.OS === 'web' ? <Image source={{ uri }} style={style} /> : <CachedImage source={{ uri }} style={style} />;
```
Comment (ipakhomov) [inline libs/shared/ui/ui-kit/external-image/component.tsx:14]: Expo image supports web as well, so we can use it directly for both platforms

File: package.json [hunk 2]
```diff
@@ -50,50 +51,52 @@
...
     "lodash": "^4.17.21",
     "luxon": "^1.28.0",
     "qs": "^6.11.0",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0",
+    "react-dom": "18.2.0",
     "react-hook-form": "^7.31.3",
-    "react-native": "0.71.8",
+    "react-native": "0.72.4",
     "react-native-expo-image-cache": "^4.1.0",
```
Comment (ipakhomov) [inline package.json:78]: This dependency can be removed

**Review by ipakhomov — CHANGES_REQUESTED** [review]
@veliseev93 we also have linting issues reported by `lint` script - could you please fix them?

**Comment by veliseev93** [comment]
@ipakhomov Done

**Review by ipakhomov — APPROVED** [review]

**Pull request merged**